File: Codes/2012-sum-of-beauty-in-the-array.py
'''
Author: LetMeFly
Date: 2025-03-11 14:04:02
LastEditors: LetMeFly.xyz
LastEditTime: 2025-03-11 14:07:14
'''
from typing import List

class Solution:
    def sumOfBeauties(self, nums: List[int]) -> int:
        mx = [0] * len(nums)
        mn = [0] * len(nums)
        now = 0
        for i in range(len(nums)):
            mx[i] = now = max(now, nums[i])
        now = 1000000
        for i in range(len(nums) - 1, -1, -1):
            mn[i] = now = min(now, nums[i])
        ans = 0
        for i in range(1, len(nums) - 1):
            if mx[i - 1] < nums[i] < mn[i + 1]:
                ans += 2
            elif nums[i - 1] < nums[i] < nums[i + 1]:
                ans += 1
        return ans

File: Codes/2012-sum-of-beauty-in-the-array.go
/*
 * @Author: LetMeFly
 * @Date: 2025-03-11 14:10:30
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-03-11 14:13:05
 */
package main

func sumOfBeauties(nums []int) (ans int) {
    mn := make([]int, len(nums))
    now := 100000
    for i := len(nums) - 1; i >= 0; i-- {
        now = min(now, nums[i])
        mn[i] = now
    }
    now = nums[0]
    for i := 1; i < len(nums) - 1; i++ {
        if now < nums[i] && nums[i] < mn[i + 1] {
            ans += 2
        } else if nums[i - 1] < nums[i] && nums[i] < nums[i + 1] {
            ans++
        }
        now = max(now, nums[i])
    }
    return
}